### PR TITLE
Add/jetpack protect

### PR DESCRIPTION
**Pull request by roccotripaldi**

### CODE REVIEW

**Comment by georgestephanis** [comment]
@roccotripaldi: I'm seeing a lot of JSHint errors on `protect.js` -- you can check by running `grunt jshint` on your branch to see them.

They're all mostly trivial, but should be tidied before merging.

**Comment by roccotripaldi** [comment]
+1 George. Thanks for catching that.

**Comment by jeffgolenski** [comment]
Note: If any styling issue appear, would be good to keep an eye on https://github.com/Automattic/jetpack/commit/c562973eae4ee963963d5bcbb987b6dbb48af636  since grunt automatically removed prefixes.

**Comment by roccotripaldi** [comment]
@lezama - i'd love your feedback on the new endpoint i built.
https://github.com/Automattic/jetpack/blob/add/jetpack-protect/json-endpoints/jetpack/class.jetpack-json-api-protect-whitelist.php

File: json-endpoints/jetpack/class.jetpack-json-api-protect-whitelist.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
+<?php
+
+class Jetpack_JSON_API_Protect_Whitelist extends Jetpack_JSON_API_Endpoint {
+	protected $needed_capabilities = 'activate_plugins';
+
+	public function callback( $path = '', $blog_id = 0, $object = null ) {
```
Comment (lezama) [inline json-endpoints/jetpack/class.jetpack-json-api-protect-whitelist.php]: I wouldn't redefine the callback method here as it's doing almost the same as the one defined in the parent class.

File: json-endpoints/jetpack/class.jetpack-json-api-protect-whitelist.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
+<?php
+
+class Jetpack_JSON_API_Protect_Whitelist extends Jetpack_JSON_API_Endpoint {
+	protected $needed_capabilities = 'activate_plugins';
```
Comment (lezama) [inline json-endpoints/jetpack/class.jetpack-json-api-protect-whitelist.php]: Shouldn't be `manage_options`?

File: json-endpoints/jetpack/class.jetpack-json-api-protect-whitelist.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+	protected function validate_input( $object ) {
+		$args = $this->input();
+		if ( ! isset( $args['whitelist'] ) || ! isset( $args['global'] ) ) {
+			return new WP_Error( 'invalid_arguments', __( 'Invalid arguments', 'jetpack' ));
+		}
+
+		$result = $this->save_whitelist( $args['whitelist'], $args['global'] );
+
+		if( ! $result ) {
```
Comment (lezama) [inline json-endpoints/jetpack/class.jetpack-json-api-protect-whitelist.php]: missing space after if

File: modules/protect/math-fallback.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		 */
+		static function math_authenticate() {
+			$salt        = get_site_option( 'jetpack_protect_key' ) . get_site_option( 'admin_email' );
+			$ans         = (int)$_POST['jetpack_protect_num'];
+			$salted_ans  = sha1( $salt . $ans );
+			$correct_ans = $_POST['jetpack_protect_answer'];
+
+			if ( ! $correct_ans && ! isset( $_POST['jetpack_protect_answer'] ) ) {
+				// TODO: Sam, we need to rethink this error message
+				wp_die( __( '<strong>This site is not properly configured.</strong> Please ask this site\'s web developer to review <a href="http://bruteprotect.com/faqs/error-bp100/">for information on how to resolve this issue</a>.', 'jetpack' ) );
```
Comment (jeherve) [inline modules/protect/math-fallback.php]: @samhotchkiss Should we start moving the BruteProtect FAQ over to jetpack.me?

Comment (samhotchkiss) [inline modules/protect/math-fallback.php]: Yep, @squirk is working on it

**Pull request merged**